RNTester/js/AccessibilityExample.js

```diff
@@ -133,6 +143,288 @@ class AccessibilityExample extends React.Component {
+    }
+    let buttonTitle = this.state.isEnabled
+      ? 'Disable selection'
+      : 'Enable selection';
+
+    return (
+      <View style={{flex: 1, flexDirection: 'row'}}>
+        <TouchableOpacity
+          ref={this.selectableElement}
+          accessible={true}
+          onPress={() => {
+            this.setState({
+              isSelected: !this.state.isSelected,
+            });
+
+            if (Platform.OS === 'android') {
+              UIManager.sendAccessibilityEvent(
+                findNodeHandle(this.selectableElement.current),
+                UIManager.AccessibilityEventTypes.typeViewClicked,
+              );
+            }
+          }}
+          accessibilityLabel="element 21"
+          accessibilityStates={accessibilityStates}
+          accessibilityHint={accessibilityHint}>
+          <Text>Selectable element example</Text>
+        </TouchableOpacity>
+        <Button
+          onPress={() => {
+            this.setState({
+              isEnabled: !this.state.isEnabled,
+            });
+          }}
+          title={buttonTitle}
+        />
+      </View>
+    );
+  }
+}
+
+class StateOnOffExample extends React.Component {
+  state = {
+    elementState: 'on',
+  };
+
+  _onElementPress = () => {
+    const elementState = this.state.elementState === 'on' ? 'off' : 'on';
+
+    this.setState({
+      elementState: elementState,
+    });
+
+    if (Platform.OS === 'android') {
+      UIManager.sendAccessibilityEvent(
+        findNodeHandle(this),
+        UIManager.AccessibilityEventTypes.typeViewClicked,
+      );
+    }
+  };
+
+  render() {
+    return (
+      <TouchableOpacity
+        onPress={this._onElementPress}
+        accessibilityLabel="element 18"
+        accessibilityStates={[this.state.elementState]}
+        accessibilityHint="click me to change state">
+        <Text>State on/off example</Text>
+      </TouchableOpacity>
+    );
+  }
+}
+
+class ExpandableElementExample extends React.Component {
+  state = {
+    expandState: 'collapsed',
+  };
+
+  _onElementPress = () => {
+    const expandState =
+      this.state.expandState === 'collapsed' ? 'expanded' : 'collapsed';
+
+    this.setState({
+      expandState: expandState,
+    });
+
+    if (Platform.OS === 'android') {
+      UIManager.sendAccessibilityEvent(
+        findNodeHandle(this),
+        UIManager.AccessibilityEventTypes.typeViewClicked,
+      );
+    }
+  };
+
+  render() {
+    return (
+      <TouchableOpacity
+        onPress={this._onElementPress}
+        accessibilityLabel="element 20"
+        accessibilityStates={[this.state.expandState]}
+        accessibilityHint="click me to change state">
+        <Text>Expandable element example</Text>
+      </TouchableOpacity>
+    );
+  }
+}
+
+class AccessibilityRoleAndStateExample extends React.Component<{}> {
+  render() {
+    return (
+      <View>
+        <View
+          accessibilityLabel="element 1"
```

Comment:
@marcmulcahy On web, we want to prefer using the element text content as the label. When a label is passed in as a prop, it's often assigned to `aria-label`, which overrides the text content. Will `accessibilityLabel` be assigned to `aria-label`? 

We've making a design decision here that we can't use `aria-labelledby` via React Native that targets web. I think we'll get by without it, but it might feel like a gap to some web developers.

Reply:
I maintain the web implementation. accessibilityLabel is already mapped to aria-label. We discussed accessibility relationships in once of the other proposals, and that will provide a way to set labelled/described on web.

Reply:
@necolas, good to know!